front/src/modules/simulationResult/components/SpeedSpaceChart/helpers.ts

```diff
@@ -0,0 +1,184 @@
+          : undefined,
+      lowerPantograph:
+        'lower_pantograph' in electrificationUsage ? electrificationUsage.lower_pantograph : false,
+    },
+  }));
+
+const formatSlopes = (slopes: PositionData<'gradient'>[]) => {
+  let actualSlopes: LayerData<number>[] = [];
+  if (slopes)
+    actualSlopes = slopes.map((slope, index) => ({
+      position: {
+        start: mToKm(index === 0 ? 0 : slopes[index - 1].position),
+        end: mToKm(slope.position),
+      },
+      value: slope.gradient,
+    }));
+  return actualSlopes;
+};
+
+const isNotCompatible = (voltage: string, profile: string) =>
+  (voltage === '1500V' && profile?.startsWith('BB')) ||
+  (voltage === '25000V' && profile?.startsWith('AA'));
+
+const getProfileValue = (
+  electricalProfileType: 'profile' | 'no_profile',
+  profile: keyof typeof electricalProfilesDesignValues | null,
+  voltage?: '1500V' | '25000V'
+) => {
+  if (electricalProfileType === 'no_profile') {
+    return { electricalProfile: 'neutral' };
+  }
+  if (profile === null || (profile && voltage && isNotCompatible(voltage, profile))) {
+    return {
+      electricalProfile: 'incompatible',
+      color:
+        voltage === '1500V'
+          ? electricalProfilesDesignValues.F.color
+          : electricalProfilesDesignValues['20000V'].color,
+    };
+  }
+  return {
+    electricalProfile: profile,
+    ...electricalProfilesDesignValues[profile!],
+  };
+};
+
+const formatElectricalProfiles = (
+  simulation: SimulationResponseSuccess,
+  electrifications: Data['electrifications']
+) => {
+  if (simulation.electrical_profiles.values.length === 0) return undefined;
+  return simulation.electrical_profiles.values.map((values, index) => {
+    const electrification = electrifications.find(
+      (electrificationValue) =>
+        electrificationValue.position.start >=
+          (index === 0 ? 0 : mmToKm(simulation.electrical_profiles.boundaries[index - 1])) &&
+        electrificationValue.position.end! <=
+          mmToKm(simulation.electrical_profiles.boundaries[index])
+    );
+
+    return {
+      position: {
+        start: mmToKm(index === 0 ? 0 : simulation.electrical_profiles.boundaries[index - 1]),
+        end: mmToKm(simulation.electrical_profiles.boundaries[index]),
+      },
+      value: getProfileValue(
+        values.electrical_profile_type,
+        'profile' in values
+          ? (values.profile as keyof typeof electricalProfilesDesignValues)
+          : null,
+        electrification?.value.voltage
+      ),
+    };
+  });
+};
+
+const formatPowerRestrictions = (powerRestrictions: SimulationPowerRestrictionRange[]) => {
+  if (powerRestrictions.length === 0) return undefined;
+  return powerRestrictions.map(({ code, handled, start, stop }) => ({
+    position: {
+      start: mToKm(start),
+      end: mToKm(stop),
+    },
+    value: {
+      powerRestriction: code,
+      handled,
+    },
+  }));
+};
```

Comment:
Instead of doing this here, you could adapt the `formatPowerRestrictionRangesWithHandled()` coming from the power restriction module and adapt the tests to fit your needs. It was only used for speedspacechart v2.